quixstreams/dataframe/series.py

```diff
@@ -9,11 +9,34 @@
 from quixstreams.core.stream.stream import Stream
 from quixstreams.models.messagecontext import MessageContext
 from .base import BaseStreaming
-from .exceptions import InvalidOperation
+from .exceptions import InvalidOperation, MissingColumn, InvalidColumnReference
 
 __all__ = ("StreamingSeries",)
 
 
+def _getitem(d: Mapping, column_name: Union[str, int]) -> object:
+    """
+    Special error handling around column referencing with SDF.
+
+    :param d: a dict-like object (usually just a dict).
+    :param column_name: the column name.
+
+    :return: Nested data from column name
+    """
+    try:
+        return d[column_name]
+    except KeyError:
+        raise MissingColumn(
+            f"Column name '{column_name}' does not exist in the StreamingDataFrame"
+        )
+    except TypeError:
+        d_type = getattr(type(d), "__name__", type(d))
+        raise InvalidColumnReference(
+            f"Cannot reference column name '{column_name}' for StreamingDataFrame "
+            f"data type '{d_type}'; a dictionary is required."
```

Comment:
Suggestion here:
> 'Cannot access column "{column_name}"; the underlying value must be a mapping, got '{d_type}' instead."

Reply:
one thought around this: how do you feel around the language with mentioning `StreamingDataFrame`?

I intentionally put that in there to make it clear where a mapping is expected (I know it should be relatively obvious based on this error message, but I wanted it unmistakable what's happening, haha)

Reply:
along that line, `underlying value` may be a bit ambiguous, potentially.

Wasn't sure the best way to refer to the DataFrame data/value, so I went with just calling it the `StreamingDataFrame`, kinda like how you'd refer to the dataframe object in Pandas.